### PR TITLE
Add documentation for VIEW privilege class 

### PR DESCRIPTION
**Pull request by mxm**

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -60,6 +60,9 @@ Breaking Changes
 Changes
 =======
 
+- Introduced a new VIEW privilege class which is used when granting/denying
+  access on views.
```
Comment (mfussenegger) [inline CHANGES.txt]: Maybe consider extending the existing view entry https://github.com/crate/crate/pull/7140/files#diff-bb3c4deeba77c2bbfd893f2e90fecc77R74 instead. Otherwise if you read this from top to bottom we mention view privileges without the introduction of views.

File: blackbox/docs/admin/privileges.rst [hunk 2]
```diff
@@ -109,15 +121,33 @@ and functions of the ``doc`` schema::
...
+tables but on the view itself, the user can still access the tables through
+the view. For example::
+
+    cr> CREATE VIEW first_customer as SELECT * from doc.accounting ORDER BY id LIMIT 1
+    CREATE OK, 1 row affected (... sec)
+
+Previously we had issued a ``DENY`` for user ``riley`` on ``doc.accounting`` but we
+can still access it through the view::
+
+    cr> Select id from first_customer;
```
Comment (mfussenegger) [inline blackbox/docs/admin/privileges.rst]: Maybe uppercase keywords like `SELECT` and `FROM` everywhere to make it consistent. Right now some are uppercased and some not.

File: blackbox/docs/admin/privileges.rst [hunk 3]
```diff
@@ -109,15 +121,33 @@ and functions of the ``doc`` schema::
...
     Privileges can be managed on all schemas and tables of the cluster,
     except the ``information_schema``.
 
+Views are on the same hierarchy with tables, i.e. a privilege on a view
+is gained through a ``GRANT`` on either the view itself, the schema the view
+belongs to, or a cluster-wide privilege. Privileges on tables which are
+referenced in the view do not grant any privileges on the view itself. On the
+contrary, even if the user does not have any privileges on a view's referenced
+tables but on the view itself, the user can still access the tables through
+the view. For example::
```
Comment (mfussenegger) [inline blackbox/docs/admin/privileges.rst]: Shouldn't we mention the owner concept here as well?

File: blackbox/docs/general/ddl/views.rst [hunk 4]
```diff
@@ -45,7 +45,8 @@ Privileges (Enterprise only)
 In order to be able to query data from a view, a user needs to have ``DQL``
 privileges on a view.
 This is the case if the user has ``DQL`` privileges on a cluster level, or on
-the schema in which the view is contained.
+the schema in which the view is contained. Privileges on tables accessed by a
```
Comment (mfussenegger) [inline blackbox/docs/general/ddl/views.rst]: Maybe rephrase this a bit..

This is the case if the user has ``DQL`` privileges on a cluster level, the schema in which the view is contained, or the view itself.

Maybe also connect the "Privileges on tables accessed by a view are not necessary" somehow better to the following paragraph that the owner needs to have permissions on the underlying relations. (I'd avoid limiting this to "tables", a view can contain other views)

File: blackbox/docs/admin/privileges.rst [hunk 5]
```diff
@@ -76,28 +76,40 @@ Hierarchical Inheritance of Privileges
...
     ...     key string,
     ...     value string
     ...   )
-    ... ) clustered by (first_column) into 5 shards;
+    ... ) clustered by (id) into 5 shards;
     CREATE OK, 1 row affected (... sec)
 
+    cr> INSERT INTO doc.accounting
+    ...   (id, second_column, third_column, fourth_column)
+    ...   VALUES (1, 'bla', 0, {key='blub', value='bluba'});
```
Comment (chaudum) [inline blackbox/docs/admin/privileges.rst]: maybe use some more meaningful fake data for the documentation.
don't like that we have bogus all over the place, but I think whenever we create new examples or touch existing, we should try to improve them

File: blackbox/docs/admin/privileges.rst (hunk 3 above)
Comment (chaudum) [reply blackbox/docs/admin/privileges.rst]: We could, but I am not sure, as it is kind of an implementation detail that is not necessarily important for the user.

Comment (mfussenegger) [reply blackbox/docs/admin/privileges.rst]: I don't think the owner is an implementation detail. It's important for users to understand the semantics. And it has a huge effect on how the privilege system behaves.

But maybe we could trim this down here a bit and link to the `ddl/views.rst` section instead.

**Comment by mxm** [comment]
@chaudum @mfussenegger Should be better now.
